aion/lib/aion/room_channel/room.ex

```diff
@@ -76,12 +89,24 @@ defmodule Aion.RoomChannel.Room do
     room.question
   end
 
-  @spec get_new_question_with_answers(integer) :: %{question: Question.t, answers: list Answer.t}
-  defp get_new_question_with_answers(room_id) do
-      question = Question.get_random_question(room_id)
-      answers = Answer.get_answers(question.id)
-      Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+  @spec get_new_question_with_answers(question_list, integer) ::  new_question_set
+  defp get_new_question_with_answers(questions, room_id) do
+    [new_question | remaining_questions] = questions
+    answers = Answer.get_answers(new_question.id)
+    Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+
+    remaining_questions = if Enum.empty?(remaining_questions), do: fetch_questions(room_id), else: remaining_questions
+
+    %{questions: remaining_questions, question: new_question, answers: answers}
+  end
+
+  @spec get_new_questions(integer) :: %{questions: question_list}
+  defp get_new_questions(room_id) do
```

Comment:
get_new_questions and fetch_questions seem are unintuitive names. Not seeing the function implementations I wouldn't be able to say what's the difference between one and the other. What this function is doing is actually `shuffling_questions`, right? Maybe that would be a better name. It would also work well in a pipeline - you could write:
```
room_id
|> fetch_questions
|> shuffle_questions
```
I think it looks very elegant and it's easy to follow.